docs/guides/4-custom-rulesets.md

```diff
@@ -343,7 +402,51 @@ aliases:
   Paths: "$.paths[*]~"
 ```
 
-Rulesets can then reference aliases in the [given](#given) keyword, either in full: `"given": "#Paths"`, or use it as a prefix for further JSON Path syntax, like dot notation: `"given": "#InfoContact.name"`.
+If you deal with a variety of different spec, you may find the above approach insufficient, particularly when the shape of the document is notably different.
+In such a case, you may want to consider using scoped aliases.
+
+```yaml
+aliases:
+  SharedParameterObject:
+    description: an optional property describing the purpose of the alias
+    targets:
+      - formats:
+          - oas2
+        given: $.parameters[*]
+      - formats:
+          - oas3
+        given: $.components.parameters[*]
+```
+
+Now, if you referenced `SharedParameterObject` alias, the chosen path would be determined based on the document you use.
+For instance, if a given document matched OpenAPI 2.x, `$.parameters[*]` would be used as the JSONPath expression.
+
+Having a closer look on the example above, one may notice that it'd be still somewhat complicated to target _all_ Parameter Objects
+that a specific OpenAPI document may contain.
+To make it more feasible and avoid overly complex JSONPath expressions, `given` can be an array.
+
+```yaml
+aliases:
+  PathItemObject: $.paths[*]
+  OperationObject: "#PathItem[get,put,post,delete,options,head,patch,trace]"
+  ParameterObject:
+    description: an optional property describing the purpose of the alias
+    targets:
+      - formats:
+          - oas2
+        given:
```

Comment:
Lol, we're ending up w similar sort of issue we always get frustrated with OpenAPI for doing... different ways to describe the same thing all over the place - e.g. an `alias` can be a string, or this object w description/targets/etc - `targets.given` can be a string or an array of strings, etc, etc 😛. No action item here, just think we should do whatever we can in the future to minimize these sorts of variants since they make tooling more difficult to build.

Reply:
Yeaaaaah. I'd just do arrays tbh, but this is now compliant with `given`